pkg/products/grafana/reconciler.go

```diff
@@ -159,16 +172,21 @@ func (r *Reconciler) ReconcileGrafanaRateLimitDashboardConfigMap(ctx context.Con
 
 	rateLimitConfigMap := &corev1.ConfigMap{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      "ratelimit-grafana-dashboard",
+			Name:      ratelimitConfigMapName,
 			Namespace: r.Config.GetNamespace(),
 		},
-		Data: map[string]string{
-			"ratelimit.json": getCustomerMonitoringGrafanaRateLimitJSON(requestsPerUnit, activeQuota),
-		},
 	}
 
 	_, err := controllerutil.CreateOrUpdate(ctx, client, rateLimitConfigMap, func() error {
+		if rateLimitConfigMap.Data == nil {
+			rateLimitConfigMap.Data = map[string]string{}
+		}
+		ratelimitJsonStr := getCustomerMonitoringGrafanaRateLimitJSON(requestsPerUnit, activeQuota)
+		if r.isRatelimitJsonChanged(ctx, client, ratelimitJsonStr) {
+			rateLimitConfigMap.Data[ratelimitCMDataKey] = ratelimitJsonStr
+		}
```

Comment:
```suggestion
		if rateLimitConfigMap.Data == nil {
			rateLimitConfigMap.Data = map[string]string{}
		}
		ratelimitJsonStr := getCustomerMonitoringGrafanaRateLimitJSON(requestsPerUnit, activeQuota)
		rateLimitConfigMap.Data[ratelimitCMDataKey] = ratelimitJsonStr
```
Actually, there's no need for additional comparison since createOrUpdate does that for us. Recommending removing the isRateLimitJsonChanged.

Reply:
done. Thanks very much for comment